roles/rhel_subscribe/tasks/enterprise.yml

```diff
@@ -2,24 +2,11 @@
 - name: Disable all repositories
   command: subscription-manager repos --disable="*"
 
-- set_fact:
-    default_ose_version: '3.6'
-  when: deployment_type == 'openshift-enterprise'
-
-- set_fact:
-    ose_version: "{{ lookup('env', 'ose_version') | default(default_ose_version, True) }}"
-
-- fail:
-    msg: "{{ ose_version }} is not a valid version for {{ deployment_type }} deployment type"
-  when:
-    - deployment_type == 'openshift-enterprise'
-    - ose_version not in ['3.1', '3.2', '3.3', '3.4', '3.5', '3.6'] )
-
 - name: Enable RHEL repositories
   command: subscription-manager repos \
                --enable="rhel-7-server-rpms" \
                --enable="rhel-7-server-extras-rpms" \
-               --enable="rhel-7-server-ose-{{ ose_version }}-rpms" \
+               --enable="rhel-7-server-ose-{{ openshift_release }}-rpms" \
```

Comment:
In this capacity, I believe we use the variable 'openshift_version' which is derived from on openshift_release.  openshift_release typically has a 'v' in front of the name.

Unfortunately, neither of this variables can be used directly.

Here's what we do in the openshift_repo role:

{{ ((openshift_release | default('')).split('.') | join(''))[0:2] }}

This is because openshift_release could contain a longer version than major/minor, eg 3.7.5.

openshift_release is created in the openshift_version role like this:

```
- set_fact:
    openshift_release: "{{ openshift_release[1:] }}"
  when:
  - openshift_release is defined
  - openshift_release[0] == 'v'
```

If this play is called before init/facts.yml, you should implement this logic here.

Reply:
Oh, right, thanks! The OpenStack playbooks do call init/fats.yml before this role but we had to do that specifically for this role. So yea, I'll make it independent of that.

Reply:
I think calling init/facts.yml is probably the easiest way forward for now.  You will ensure you're not missing out on any changes there.

You will probably still need this part:

{{ ((openshift_release | default('')).split('.') | join(''))[0:2] }}

Which might need to be modified to include the '.' in the last join if the repo's contain a '.' character in the version number:

{{ ((openshift_release | default('')).split('.') | join('.'))[0:2] }}

Reply:
I used a slightly modified version that will work with 2-digit minor versions as well (e.g. 3.10):

    {{ (openshift_release | default('')).split('.')[0:2] | join('.') }}